app/src/main/java/com/bogdan/codeforceswatcher/feature/users/redux/UsersReducer.kt

```diff
@@ -0,0 +1,38 @@
+package com.bogdan.codeforceswatcher.feature.users.redux
+
+import com.bogdan.codeforceswatcher.feature.users.redux.actions.SortActions
+import com.bogdan.codeforceswatcher.feature.users.redux.request.UsersRequests
+import com.bogdan.codeforceswatcher.redux.AppState
+import org.rekotlin.Action
+
+fun usersReducer(action: Action, state: AppState): UsersState {
+    var newState = state.users
+
+    when (action) {
+        is UsersRequests.FetchUsers -> {
+            newState = newState.copy(
+                status = UsersState.Status.PENDING
+            )
+        }
+        is UsersRequests.FetchUsers.Success -> {
+            newState = newState.copy(
+                status = UsersState.Status.IDLE,
+                users = Sorting.sort(action.users, newState.sortType),
+                result = action.result
+            )
+        }
+        is UsersRequests.FetchUsers.Failure -> {
+            newState = newState.copy(
+                status = UsersState.Status.IDLE
+            )
+        }
+        is SortActions.Sort -> {
+            newState = newState.copy(
+                users = Sorting.sort(newState.users, action.sortType),
```

Comment:
Then it can be called like `newState.users.sort(action.sortType)`.

Reply:
And it seems to be UI-related part, so move it to `newState` in `ContestsFragment`.